src/mon/MonCap.cc

```diff
@@ -258,6 +258,17 @@ void MonCapGrant::expand_profile_mon(const EntityName& name) const
     profile_grants.back().command_args["caps_osd"] = StringConstraint(
       StringConstraint::MATCH_TYPE_EQUAL, "allow rwx");
   }
+  if (profile == "bootstrap-rbd") {
+    profile_grants.push_back(MonCapGrant("mon", MON_CAP_R));  // read monmap
+    profile_grants.push_back(MonCapGrant("auth get-or-create"));  // FIXME: this can expose other mds keys
```

Comment:
In the future the way to close this is to have the client generate the keys and pass them to the mon. That way the reply from the mon is simply success/failure and there is no potential leak of info.